static/js/components/Loading_test.js

```diff
@@ -1,134 +1,44 @@
-          processing: false
-        },
-        {
-          loaded:     false,
-          processing: false
-        }
-      ])
-      assert.equal(wrapper.text(), "Loading")
-    })
-
-    it("single state with processing:true", () => {
-      const wrapper = renderLoading([
-        {
-          loaded:     false,
-          processing: true
-        }
-      ])
-      assert.equal(wrapper.text(), "Loading")
-    })
-
-    it("multiple states with one processing:true", () => {
-      const wrapper = renderLoading([
-        {
-          loaded:     false,
-          processing: false
-        },
-        {
-          loaded:     false,
-          processing: true
-        }
-      ])
+    it("render loading if not loaded and not errored", () => {
+      const wrapper = renderLoading(false, false)
       assert.equal(wrapper.text(), "Loading")
     })
   })
 
   describe("should render errors correctly", () => {
-    it("single state with error", () => {
-      const wrapper = renderLoading([
-        {
-          loaded:     true,
-          processing: false,
-          error:      "bad things"
-        }
-      ])
-      assert.equal(wrapper.text(), "Error loading page")
-    })
-
-    it("multiple states all with errors", () => {
-      const wrapper = renderLoading([
-        {
-          loaded:     true,
-          processing: false,
-          error:      "bad things"
-        },
-        {
-          loaded:     true,
-          processing: false,
-          error:      "bad things"
-        }
-      ])
+    it("if not loaded", () => {
+      const wrapper = renderLoading(false, true)
       assert.equal(wrapper.text(), "Error loading page")
     })
 
-    it("multiple states with one error", () => {
-      const wrapper = renderLoading([
-        {
-          loaded:     true,
-          processing: false,
-          error:      undefined
-        },
-        {
-          loaded:     true,
-          processing: false,
-          error:      "bad things"
-        }
-      ])
+    it("if loaded", () => {
+      const wrapper = renderLoading(true, true)
       assert.equal(wrapper.text(), "Error loading page")
     })
   })
 
-  describe("should the contents if no errors", () => {
+  describe("should the contents if no errors and loaded", () => {
     it("single state", () => {
-      const wrapper = renderLoading([
-        {
-          loaded:     true,
-          processing: false,
-          error:      undefined
-        }
-      ])
-      assert.equal(wrapper.text(), "CONTENT")
-    })
-
-    it("multiple states", () => {
-      const wrapper = renderLoading([
-        {
-          loaded:     true,
-          processing: false,
-          error:      undefined
-        },
-        {
-          loaded:     true,
-          processing: false,
-          error:      undefined
-        }
-      ])
+      const wrapper = renderLoading(true, false)
       assert.equal(wrapper.text(), "CONTENT")
     })
   })
```

Comment:
Is there a test that exercises the `updateRequirements` callback? if not I think it would be good to add a simple one that just ensures that a `sinon.stub` gets called.